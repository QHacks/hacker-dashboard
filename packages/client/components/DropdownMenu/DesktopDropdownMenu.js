import React, { Component } from "react";
import { blue, blueLight } from "../../assets/colors";
import { linkExternalTrusted } from "../../assets/constants";

class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dropdownShown: false
    };
  }

  render() {
    return (
      <div onMouseLeave={() => this.setState({ dropdownShown: false })}>
        <button
          css={`
            color: ${blue};
            font-weight: ${this.state.dropdownShown ? "bold" : 600};
            border: none;
          `}
          onMouseOver={() => this.setState({ dropdownShown: true })}
        >
          {this.props.title}
          {"  "}
          {this.state.dropdownShown ? "\u25B2" : "\u25BC"}
        </button>
        {this.state.dropdownShown ? (
          <div
            css={`
              background-color: white;
              border-top: 2px solid ${blue};
              padding: 8px 0;
              border-radius: 0 0 6px 6px;
              box-shadow: 0 4px 12px 0 rgba(167, 176, 194, 0.4);
              > a {
                display: block;
                line-height 1;
                padding: 12px !important;
                margin: 0;
                border-bottom: none !important;
                &:hover {
                  color: ${blueLight};
                }
              }
            `}
          >
            {this.props.links.map((link) => (
              <a
                {...linkExternalTrusted}
                key={`${link.href}-${link.text}`}
                href={link.href}
              >
                {link.text}
              </a>
            ))}
          </div>
        ) : (
          ""
        )}
      </div>
    );
  }
}

export default Dropdown;
