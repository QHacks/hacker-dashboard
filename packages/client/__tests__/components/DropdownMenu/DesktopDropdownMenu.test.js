import React from "react";
import DesktopDropdownMenu from "../../../components/DropdownMenu/DesktopDropdownMenu";
import { shallow } from "enzyme";

describe("DesktopDropdownMenu", () => {
  it("renders correctly", () => {
    expect(shallow(<DesktopDropdownMenu />)).toMatchSnapshot();
  });
});
