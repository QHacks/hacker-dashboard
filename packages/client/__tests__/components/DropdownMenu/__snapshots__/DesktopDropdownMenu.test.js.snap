// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopDropdownMenu renders correctly 1`] = `
<div
  onMouseLeave={[Function]}
>
  <button
    className="css-mrl3g0"
    onMouseOver={[Function]}
  >
      
    â–¼
  </button>
</div>
`;
